Resetting the MiSTer causes autostart to never evaluate

When the autosplitter is waiting for the autostart condition to be met, if the user resets the MiSTer then the auto splitter goes into an error state and won't attempt to evaluate the autostart condition anymore.

```sh
$ ./sni-autosplitter --sni-host 192.168.0.17                      
┌─────────────────────────────────────────────────────────────┐
│                    SNI AutoSplitter                         │
│          LiveSplit One + Super Nintendo Interface           │
└─────────────────────────────────────────────────────────────┘

[INFO] Discovering run configurations...
[SUCCESS] Found 2 run configuration(s)
[INFO] Available runs:
  1. A Link to the Past - Any% No Major Glitches (alttp - 15 splits)
  2. A Link to the Past - Master Sword No Major Glitches (alttp - 5 splits)

Select run (1-2) or 'q' to quit: 2
[SUCCESS] Selected: Master Sword No Major Glitches
[INFO] Loading game configuration...
[SUCCESS] Loaded:  - Master Sword No Major Glitches (5 splits)
[INFO] Connecting to SNI server...
[SUCCESS] Connected to SNI server
[INFO] Discovering SNI devices...
[SUCCESS] Found 1 compatible device(s)
[INFO] Found device: MiSTer SNES (mister)
Use this device? (y/n): y
[INFO] Testing device connection...
[SUCCESS] Device connection test successful
[INFO] Device Information:
  Name: MiSTer SNES
  Type: mister
  URI: mister:://mister
  Capabilities: Memory Read, Memory Write
[INFO] Detecting memory mapping...
[WARNING] Memory mapping detection failed, using LoROM: memory mapping detection failed: rpc error: code = Unimplemented desc = Reading from ROM is unsupported
[SUCCESS] Memory mapping: LoROM
[SUCCESS] SNI integration complete!
[INFO] Ready to autosplit:  - Master Sword No Major Glitches
[INFO] Device: MiSTer SNES
[INFO] Memory mapping: LoROM
[INFO] Initializing splitting engine...
[INFO] Initializing splitting engine...
[SUCCESS] Splitting engine initialized
[INFO] Starting splitting engine...
[INFO] Starting LiveSplit server...
[SUCCESS] LiveSplit server started
[SUCCESS] Splitting engine started

────────────────────────────────────────────────────────────
Engine Status: Waiting for Start
Run: A Link to the Past - Master Sword No Major Glitches (The Legend of Zelda: A Link to the Past)
────────────────────────────────────────────────────────────
[INFO] Entering interactive mode...
[INFO] You can now control the autosplitter or let it run automatically
[INFO] Entering interactive mode

Available commands:
  h, help    - Show this help
  s, status  - Show current status
  stats      - Show detailed statistics
  q, quit    - Exit interactive mode

Command (h for help): [ERROR] ❌ Error: failed to evaluate autostart condition: failed to read memory at 0xF50010: failed to read memory at 0xF50010: rpc error: code = Internal desc = The SNES core is not connected
```

--- a/internal/engine/engine.go
+++ b/internal/engine/engine.go
@@ -12,6 +12,11 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// Primitive manual testing of resetting a MiSTer has a threshold of about 1.25s where it
+// will fail to connect to the device after a reset. This default gives an extra second of
+// tolerance to account for other transient issues that may arise.
+const DefaultAutostartErrorTolerance = 2*time.Second + 250*time.Millisecond
+
 // SplittingEngine manages the autosplitting logic and state
 type SplittingEngine struct {
 	logger    *logrus.Logger
@@ -21,13 +26,15 @@ type SplittingEngine struct {
 	session   *SplitterSession
 
 	// Configuration
-	pollInterval time.Duration
+	pollInterval            time.Duration
+	autostartErrorTolerance time.Duration
 
 	// Runtime state
-	running bool
-	ctx     context.Context
-	cancel  context.CancelFunc
-	mu      sync.RWMutex
+	running             bool
+	ctx                 context.Context
+	cancel              context.CancelFunc
+	mu                  sync.RWMutex
+	autostartErrorStart time.Time
 
 	// Event subscribers
 	splitSubscribers  map[chan SplitEvent]struct{}
@@ -103,12 +110,16 @@ func (se *SplittingEngine) tick() error {
 
 // checkAutostart checks if the autostart condition is met
 func (se *SplittingEngine) checkAutostart() error {
-	autostart := se.session.GetGameConfig().Autostart
-	autostartState := se.session.GetAutostartState()
+	// pick up the configured tolerance or fall back to the default
+	tol := se.autostartErrorTolerance
+	if tol == 0 {
+		tol = DefaultAutostartErrorTolerance
+	}
 
-	// Errors are intentionally ignored here. Resetting the SNES can cause errors due to memory not
-	// being readable, but ignoring the error allows the run loop to continue trying.
-	result, _ := se.evaluator.EvaluateComplexCondition(se.ctx, se.device.URI, &autostart, autostartState)
+	result, err := se.evaluateAutostartWithTolerance(tol)
+	if err != nil {
+		return fmt.Errorf("failed to evaluate autostart condition: %w", err)
+	}
 	if result {
 		se.logger.Info("Autostart condition met - starting run")
 		se.session.Start()
@@ -119,6 +130,30 @@ func (se *SplittingEngine) checkAutostart() error {
 	}
 
 	return nil
+}
+
+// evaluateAutostartWithTolerance suppresses errors until they persist beyond the tolerance window
+func (se *SplittingEngine) evaluateAutostartWithTolerance(tol time.Duration) (bool, error) {
+	autostart := se.session.GetGameConfig().Autostart
+	autostartState := se.session.GetAutostartState()
+
+	result, err := se.evaluator.EvaluateComplexCondition(se.ctx, se.device.URI, &autostart, autostartState)
+	if err != nil {
+		if se.autostartErrorStart.IsZero() {
+			se.autostartErrorStart = time.Now()
+		}
+		if time.Since(se.autostartErrorStart) > tol {
+			// reset so we don’t immediately re-error next tick
+			se.autostartErrorStart = time.Time{}
+			return false, err
+		}
+		// still within tolerance → suppress
+		return false, nil
+	}
+
+	// success → reset any previous error timestamp
+	se.autostartErrorStart = time.Time{}
+	return result, nil
 }
 
 // checkCurrentSplit checks if the current split condition is met

--- a/internal/engine/engine.go
+++ b/internal/engine/engine.go
@@ -15,7 +15,7 @@ import (
 // Primitive manual testing of resetting a MiSTer has a threshold of about 1.25s where it
 // will fail to connect to the device after a reset. This default gives an extra second of
 // tolerance to account for other transient issues that may arise.
-const DefaultAutostartErrorTolerance = 2*time.Second + 250*time.Millisecond
+const DefaultErrorTolerance = 2*time.Second + 250*time.Millisecond
 
 // SplittingEngine manages the autosplitting logic and state
 type SplittingEngine struct {
@@ -26,15 +26,15 @@ type SplittingEngine struct {
 	session   *SplitterSession
 
 	// Configuration
-	pollInterval            time.Duration
-	autostartErrorTolerance time.Duration
+	pollInterval   time.Duration
+	errorTolerance time.Duration
 
 	// Runtime state
-	running             bool
-	ctx                 context.Context
-	cancel              context.CancelFunc
-	mu                  sync.RWMutex
-	autostartErrorStart time.Time
+	running    bool
+	ctx        context.Context
+	cancel     context.CancelFunc
+	mu         sync.RWMutex
+	errorStart time.Time
 
 	// Event subscribers
 	splitSubscribers  map[chan SplitEvent]struct{}
@@ -110,13 +110,10 @@ func (se *SplittingEngine) tick() error {
 
 // checkAutostart checks if the autostart condition is met
 func (se *SplittingEngine) checkAutostart() error {
-	// pick up the configured tolerance or fall back to the default
-	tol := se.autostartErrorTolerance
-	if tol == 0 {
-		tol = DefaultAutostartErrorTolerance
-	}
+	autostart := se.session.GetGameConfig().Autostart
+	autostartState := se.session.GetAutostartState()
 
-	result, err := se.evaluateAutostartWithTolerance(tol)
+	result, err := se.evaluateComplexConditionWithTolerance(&autostart, autostartState)
 	if err != nil {
 		return fmt.Errorf("failed to evaluate autostart condition: %w", err)
 	}
@@ -130,30 +127,6 @@ func (se *SplittingEngine) checkAutostart() error {
 	}
 
 	return nil
-}
-
-// evaluateAutostartWithTolerance suppresses errors until they persist beyond the tolerance window
-func (se *SplittingEngine) evaluateAutostartWithTolerance(tol time.Duration) (bool, error) {
-	autostart := se.session.GetGameConfig().Autostart
-	autostartState := se.session.GetAutostartState()
-
-	result, err := se.evaluator.EvaluateComplexCondition(se.ctx, se.device.URI, &autostart, autostartState)
-	if err != nil {
-		if se.autostartErrorStart.IsZero() {
-			se.autostartErrorStart = time.Now()
-		}
-		if time.Since(se.autostartErrorStart) > tol {
-			// reset so we don’t immediately re-error next tick
-			se.autostartErrorStart = time.Time{}
-			return false, err
-		}
-		// still within tolerance → suppress
-		return false, nil
-	}
-
-	// success → reset any previous error timestamp
-	se.autostartErrorStart = time.Time{}
-	return result, nil
 }
 
 // checkCurrentSplit checks if the current split condition is met
@@ -177,7 +150,8 @@ func (se *SplittingEngine) checkCurrentSplit() error {
 	splitState := se.session.GetSplitState(currentSplitName)
 
 	// Evaluate the split condition
-	result, err := se.evaluator.EvaluateComplexCondition(se.ctx, se.device.URI, split, splitState)
+	// apply tolerance logic for current split
+	result, err := se.evaluateComplexConditionWithTolerance(split, splitState)
 	if err != nil {
 		return fmt.Errorf("failed to evaluate split condition for '%s': %w", currentSplitName, err)
 	}
@@ -188,6 +162,32 @@ func (se *SplittingEngine) checkCurrentSplit() error {
 	}
 
 	return nil
+}
+
+// evaluateComplexConditionWithTolerance suppresses errors until they persist beyond the tolerance window
+func (se *SplittingEngine) evaluateComplexConditionWithTolerance(split *config.Split, splitState *SplitState) (bool, error) {
+	result, err := se.evaluator.EvaluateComplexCondition(se.ctx, se.device.URI, split, splitState)
+	if err != nil {
+		tol := se.errorTolerance
+		if tol == 0 {
+			tol = DefaultErrorTolerance
+		}
+
+		if se.errorStart.IsZero() {
+			se.errorStart = time.Now()
+		}
+		if time.Since(se.errorStart) > tol {
+			// reset so we don’t immediately re-error next tick
+			se.errorStart = time.Time{}
+			return false, err
+		}
+		// still within tolerance → suppress
+		return false, nil
+	}
+
+	// success → reset any previous error timestamp
+	se.errorStart = time.Time{}
+	return result, nil
 }
 
 // triggerSplit triggers a split and handles the transition

--- a/internal/engine/engine.go
+++ b/internal/engine/engine.go
@@ -106,11 +106,9 @@ func (se *SplittingEngine) checkAutostart() error {
 	autostart := se.session.GetGameConfig().Autostart
 	autostartState := se.session.GetAutostartState()
 
-	result, err := se.evaluator.EvaluateComplexCondition(se.ctx, se.device.URI, &autostart, autostartState)
-	if err != nil {
-		return fmt.Errorf("failed to evaluate autostart condition: %w", err)
-	}
-
+	// Errors are intentionally ignored here. Resetting the SNES can cause errors due to memory not
+	// being readable, but ignoring the error allows the run loop to continue trying.
+	result, _ := se.evaluator.EvaluateComplexCondition(se.ctx, se.device.URI, &autostart, autostartState)
 	if result {
 		se.logger.Info("Autostart condition met - starting run")
 		se.session.Start()

